### PR TITLE
Add current_user_guid to service unbind

### PR DESCRIPTION
**Pull request by suhlig**
Fixes [#151431663](https://www.pivotaltracker.com/n/projects/2105761/stories/151431663). This PR is based on a collaboration with the SAPI team (@samze).

* [X] I have viewed signed and have submitted the Contributor License Agreement

* [X] I have made this pull request to the `master` branch

* [X] I have run all the unit tests using `bundle exec rake`

* [X] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats) on bosh lite

### CODE REVIEW

File: lib/services/service_brokers/v2/http_client.rb [hunk 1]
```diff
@@ -86,22 +92,22 @@ def initialize(attrs, logger=nil)
       end
 
       def get(path)
-        make_request(:get, uri_for(path), nil, nil)
+        make_request(:get, uri_for(path))
       end
```
Comment (ljfranklin) [inline lib/services/service_brokers/v2/http_client.rb]: @suhlig prior to this PR, we were setting the `HEADER_BROKER_API_ORIGINATING_IDENTITY` for every HTTP method (`GET`, `PUT`, `PATCH`, `DELETE`), but this PR only sets the user context for `DELETE`. Was this intentional?

Comment (suhlig) [reply lib/services/service_brokers/v2/http_client.rb]: I refactored the `make_request` method to default to `body = nil` and take additional options (incl. the former `content_type`); all defaulting to `nil`, too. The reason for this refactoring was either Rubocop or Codeclimate (don't remember which one). I believe the resulting behaviour is the same.

**Comment by suhlig** [comment]
I should have been more clear in the initial description of this PR: For calling with `async=true`, this PR fixes the passing of `HEADER_BROKER_API_ORIGINATING_IDENTITY` only for `DELETE`.

File: lib/services/service_brokers/v2/http_client.rb (hunk 1 above)
Comment (ljfranklin) [reply lib/services/service_brokers/v2/http_client.rb]: @suhlig in the previous implementation, `make_request` would try to set the `HEADER_BROKER_API_ORIGINATING_IDENTITY` for `GET`, `PUT`, `PATCH`, and `DELETE` by pulling it off the `VCAP::CloudController::SecurityContext`.

While I like the change to instead pass in the user guid prior to enqueuing the job, I'm concerned that the new code no longer sets the `HEADER_BROKER_API_ORIGINATING_IDENTITY` header for `GET`, `PUT`, or `PATCH`. Seems like the user guid should also be passed to the other HTTP methods, not just `DELETE`.

Comment (ljfranklin) [reply lib/services/service_brokers/v2/http_client.rb]: @suhlig @Samze, this story is still towards the top of our backlog. We can get it merged once you've addressed the above comment.

Comment (suhlig) [reply lib/services/service_brokers/v2/http_client.rb]: I _think_ the implementation of `make_request` in this PR behaves the same as before.

*   In master, we always read the `user_guid` from the security context:

    ```ruby
    user_guid = VCAP::CloudController::SecurityContext.current_user_guid
    opts[:header][VCAP::Request::HEADER_BROKER_API_ORIGINATING_IDENTITY] = originating_identity(user_guid) if user_guid
    ```

*   In this PR, we read `user_guid` from `options[:user_guid]`, and if that one is nil, we fall back to the security context as before (see the implementation of the `user_guid` method below):

    ```ruby
    user_guid = user_guid(options)
    opts[:header][VCAP::Request::HEADER_BROKER_API_ORIGINATING_IDENTITY] = IdentityEncoder.encode(user_guid) if user_guid
    ...
    def user_guid(options)
      if options[:user_guid]
        options[:user_guid]
      else
        VCAP::CloudController::SecurityContext.current_user_guid
      end
    end
    ```

Perhaps I am overlooking something else? I am happy to jump on a screen sharing session in your morning.

Comment (Samze) [reply lib/services/service_brokers/v2/http_client.rb]: @ljfranklin sorry for the delay. The [previous pr](https://github.com/cloudfoundry/cloud_controller_ng/pull/877) added the originating header to outgoing broker requests. It achieved this by reading the user information from the SecurityContext. It purposefully did not send user information when a task was asynchronous, as we considered it to be the "platform" making the request for things like last operation.

However, the async=true flag for unbind service creates an async job which would not have the user information. @suhlig's PR adds this info explicitly on the call to cover the async case.

Deprovision has the same issue which we discussed addressing in a different PR.

**Comment by ljfranklin** [comment]
@suhlig whoops, not sure what I was seeing when I first reviewed the PR. Thanks for the clarification, merging now!

**Pull request merged**